contrib/dracut/90zfs/mount-zfs.sh.in

```diff
@@ -60,24 +60,11 @@ if import_pool "${ZFS_POOL}" ; then
 	if [ $(zpool list -H -o feature@encryption $(echo "${ZFS_POOL}" | awk -F\/ '{print $1}')) == 'active' ]; then
 		# if the root dataset has encryption enabled
 		if $(zfs list -H -o encryption "${ZFS_DATASET}" | grep -q -v off); then
```

Comment:
I was thinking that this and the subsequent 'zfs get' would be combined into a single 'zfs list -H -o encryptionroot' that would be non-empty if encryption is enabled as well as providing us a path to unlock immediately.

Reply:
I'm personally opposed to parsing command output that isn't extremely simple (as in just the data you want).  It also adds code complexity to the scripts.  I'll do it if others want, but I generally don't like it.

Reply:
there is no parsing required, `zfs get -H -o value encryptionroot` would print out *only* what we want.

Reply:
```
# zfs get -H -o value encryptionroot rpool/ocean/GENTOO/ROOT
rpool/ocean
# zfs get -H -o value encryptionroot rpool                  
-
```

just check that it is not `-` and encryption is enabled plus we'll have the path to unlock.

Reply:
so something like this

`if [[ $(zfs get -H -o value encryptionroot ${ZFS_DATASET}) | grep -v '^-$' | grep -v off ]]; then`

or

```
ENCRYPTIONROOT=$(zfs get -H -o value encryptionroot ${ZFS_DATASET})
if [[ $(echo -n "${ENCRYPTIONROOT}" | grep -v '^-$' | grep -v off ]]; then
  TRY_COUNT=5
  while [ $TRY_COUNT != 0 ]; do
    zfs load-key "$(${ENCRYPTIONROOT})"
    [ $? == 0 ] && break
    ((TRY_COUNT-=1))
  done
fi

Reply:
the 2nd one is how I thought it'd look, whichever one has the least number of `zfs` commands wins

Reply:
ya, that's more the target